src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyHandlerImpl.java

```diff
@@ -57,8 +57,7 @@ public ProxyResponseResult validateProxyResponse(ByteBuffer buffer) {
             if (responseScanner.hasNextLine()) {
                 final String firstLine = responseScanner.nextLine();
                 if ((firstLine.toLowerCase().contains("http/1.1") || firstLine.toLowerCase().contains("http/1.0"))
-                        && firstLine.contains("200")
-                        && firstLine.toLowerCase().contains("connection established")) {
+                        && firstLine.matches("(.*)2[0-9][0-9](.*)")) {

```

Comment:
I don't believe this is the correct fix. The RFC defines a format for the first line, the status-line. I would update the verification of this to match that.
https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html

Reply:
I disagree, you can read on tehe RFC:

"The individual values of the numeric status codes defined for HTTP/1.1, and an example set of corresponding Reason-Phrase's, are presented below. The reason phrases listed here are only recommendations -- they MAY be replaced by local equivalents without affecting the protocol."

The reason codes are something that can be implemented without affecting the protocol as per the specification, so adding a list seems to be the wrong solution. The true fix should be to look at the codes as per the protocol.

Reply:
the current tests already work, since they don't validate the reason phrase

Reply:
https://github.com/ataquedosesquilos/qpid-proton-j-extensions/blob/master/src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyHandlerImpl.java

https://github.com/ataquedosesquilos/qpid-proton-j-extensions/blob/master/src/test/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyHandlerImplTest.java

Created a new version and tested it to be working.

                if (firstLine.toLowerCase().trim().matches("^http\\/(1.0|1.1)\\s(200|201|202|203|204|205|206)\\s\\w.*$")) {

this regex should do the trick, validates the http version, all status codes allowed on the rfc and makes sure there's a reason phrase, i don't have access so i can't do a pr myself.
Added some unit tests also for all http error codes and in case we get an empty reason phrase it fails also.


can this be included on a release soon? we need to go live with this functionality, currently we are done with development but we can't go live with a fork that is not officially released by microsoft.